pkg/clients/argocd.go

```diff
@@ -67,6 +68,14 @@ func UseProviderConfig(ctx context.Context, c client.Client, mg resource.Managed
 		return nil, errors.Wrap(err, "cannot track ProviderConfig usage")
 	}
 
+	insecure := false
+	if pc.Spec.Insecure != nil {
+		insecure = *pc.Spec.Insecure
+	}
+	plaintext := false
+	if pc.Spec.Insecure != nil {
```

Comment:
```suggestion
	if pc.Spec.PlainText != nil {
```

Reply:
The defaults should also be documented in https://github.com/crossplane-contrib/provider-argocd/blob/main/apis/v1alpha1/providerconfig_types.go#L30 and https://github.com/crossplane-contrib/provider-argocd/blob/main/apis/v1alpha1/providerconfig_types.go#L34

Reply:
Ooh! Thanks for catching that.